tests/test_cli_compile.py

```diff
@@ -691,8 +691,7 @@ def test_annotate_option(pip_conf, runner, option, expected):
 
 
 @pytest.mark.parametrize(
-    "option, expected",
-    [("--allow-unsafe", "\nsetuptools=="), (None, "\n# setuptools==")],
+    "option, expected", [("--allow-unsafe", "\nsetuptools=="), (None, "\n# setuptools")]
```

Comment:
I don't see any difference in this change. Looks unrelated...

Reply:
Oh, I see. The second one doesn't have `==`. It's been hard to spot. I suggest that you minimize the diff in order to make it review-friendly.

Reply:
Yeah, that's why I've split the PR into two commits.

Reply:
Well, it's not visible in the PR diff and folks mostly don't do reviews per commit (I do sometimes but to all the time).

OTOH, you mix up two separate tasks in this PR:
1) reformatting the code
2) changing things

This makes it non-atomic, which is not good in my book. As you can see, this confuses even me. And I do a lot more reviews than most of the regular folks (compared to other types of contributions). Which opens up possibilities to sneak in a few more bugs unnoticed. Humans are error-prone and byte-by-byte comparisons are hard without any highlighting.

Reply:
Yeah, it should have been separate PRs. I thought these changes would be easy to review (which is apparently not). My bad. Would you like me to split this into two PRs or we could go on?

Reply:
Yes, it would be better to avoid reformatting things in PRs that target changing the behavior.

Reply:
I must note the first commit is not about reformatting. It improves the test. The actual test was not complete enough to make changes.

Reply:
"Yes" means split or go on?

Reply:
Please split

Reply:
P.S. these comments refer to the second commit which both (1) reformats 2 separate lines into one **and** (2) changes what's needed for the test. Hence two atomic changes in a single commit.

Reply:
We use black. Reformat is necessary, but not the intention.

Reply:
Damn. black again...

Reply:
> Please split

okay 👍

Reply:
First one #986.

Reply:
I've rebased the branch onto the master.

Reply:
Wait, it still has reformatting and a useful change smashed in together.

Reply:
Because first, we do useful change, then black reformats it. And we can't split it, because of the linter.

Reply:
And that's bad because the change doesn't show up highlighted in diff and it looks like you just took two lines and merged them into one because of this making the review next to impossible.

Reply:
> you just took two lines and merged them into

this is what black does, not me :)

Reply:
you could always `# black off` to work around black's downsides

Reply:
I mean, if this keeps happening, we need to know how to make things review-friendly. Otherwise, we'll just silently accumulate bugs.

Reply:
This is definitely what I don't want to do. Better to not use black at al then.

Reply:
It would be nice if github's diff-tool would highlight changed string (which is removed "==" in this particular case).

Reply:
> I mean, if this keeps happening, we need to know how to make things review-friendly. Otherwise, we'll just silently accumulate bugs.

Just wondering is there any discussion about black and non-friendly-review changes.  If not, let's open the discussion in black's issue-tracker for a start?

Reply:
Yes, I usually advocate for not using black because of its problems and freedoms it steels but it's not my call to make here, I guess :)

Regarding diff. I don't think that `git diff` as well as the default `diff` tool is capable of detecting string merge *and* edit. To these tools, it just looks like removing one line and editing another.

OTOH if you look to this diff line https://github.com/jazzband/pip-tools/pull/975/files#diff-9960f0e0139a2d6000305af0c80dac37R194 you'll see a part of the string with highlighted _which is what our goal should be_.

I dream of a tool that works similar to black but only reformats things that violate PEP8 and not everything unrelated to the target change...

Reply:
> Just wondering is there any discussion about black and non-friendly-review changes. If not, let's open the discussion in black's issue-tracker for a start?

Not that I know off. But whenever I start it, everybody starts hating me for this. So I'm not very enthusiastic about participating in yet another argument personally...

Reply:
> OTOH if you look to this diff line https://github.com/jazzband/pip-tools/pull/975/files#diff-9960f0e0139a2d6000305af0c80dac37R194 you'll see a part of the string with highlighted _which is what our goal should be_.

That's what I'm talking about!

> Not that I know of. But whenever I start it, everybody starts hating me for this. So I'm not very enthusiastic about participating in yet another argument personally...

Haha! That's a rough path 😄

Reply:
Anyway. I think that one commit should have a logic change and then autoformatting should be a separate commit.
Or just merge it if you want.

Reply:
That makes sense, but as you mentioned before:
> Well, it's not visible in the PR diff and folks mostly don't do reviews per commit (I do sometimes but to all the time).
 
And I can't merge this unless someone approves it.

Reply:
Oh, that approval from @PeterJCLaw doesn't work..

Reply:
> as you mentioned before

I'd like to point out that both individual commits and PRs can be atomic, they just refer to different abstraction levels.